h2o-algos/src/main/java/hex/gram/Gram.java

```diff
@@ -758,6 +758,39 @@ public void mul(double [] x, double [] res){
     }
   }
 
+  /*
+  This method will not allocate the extra memory and hence is considered for lowMemory systems.
+  However, need to consider case when you have categoricals!  Make them part of the matrix
+  in the multiplication process.
+   */
```

Comment:
There should not be multiple multiplication methods unless we really need them.
You should either drop this one or replace the older method.